src/components/calculator.js

```diff
@@ -1,37 +1,61 @@
-function Calculator() {
+import React, { useState } from 'react';
+import Button from './myButton';
+import calculate from '../logic/calculate';
+
+const Calculator = () => {
+  const [display, setDisplay] = useState('0');
+  const [calculatorData, setCalculatorData] = useState({});
+  const handleButtonClick = (buttonName) => {
+    const newData = calculate(calculatorData, buttonName);
+    if (buttonName === '=') {
+      if (newData.total) {
+        setDisplay(newData.total);
+      } else {
+        setDisplay('0');
+      }
+      setCalculatorData({});
+    } else if (['+', '-', 'x', '÷'].includes(buttonName)) {
+      setDisplay(buttonName);
+      setCalculatorData({
+        total: display,
+        next: null,
+        operation: buttonName,
+      });
+    } else {
+      if (newData.next) {
+        setDisplay(newData.next);
+      } else {
+        setDisplay('0');
+      }
+      setCalculatorData(newData);
+    }
+  };
   return (
     <div className="container">
-      <div className="wraper">
-
-        <div className="screen">0</div>
-
-        <div className="btn light-gray">AC</div>
-        <div className="btn light-gray">+/-</div>
-        <div className="btn light-gray">%</div>
-        <div className="btn orange">/</div>
-
-        <div className="btn">7</div>
-        <div className="btn">8</div>
-        <div className="btn">9</div>
-        <div className="btn orange">x</div>
-
-        <div className="btn">4</div>
-        <div className="btn">5</div>
-        <div className="btn">6</div>
-        <div className="btn orange">-</div>
-
-        <div className="btn">1</div>
-        <div className="btn">2</div>
-        <div className="btn">3</div>
-        <div className="btn orange">+</div>
-
-        <div className="btn">0</div>
-        <div className="btn" />
-        <div className="btn">.</div>
-        <div className="btn orange">=</div>
-
+      <div className="wrapper">
+        <div className="screen">{display}</div>
+        <Button className="btn light-gray" onClick={() => handleButtonClick('AC')}>AC</Button>
+        <Button className="btn light-gray" onClick={() => handleButtonClick('+/-')}>+/-</Button>
+        <Button className="btn light-gray" onClick={() => handleButtonClick('%')}>%</Button>
+        <Button className="btn orange" onClick={() => handleButtonClick('/')}>/</Button>
```

Comment:
- [ ] when I try to use division I get an error, kindly fix it 
![Screenshot from 2023-05-31 21-00-54](https://github.com/amanuelbelete95/math-magician-one/assets/66097371/c8dccc24-ad0a-4f4b-a633-51ba05068217)